### PR TITLE
Review Altair core epoch files with changes

### PR DESCRIPTION
**Pull request by terencechain**
All of the changes are cosmetic except for one. Change list:
 - If update sync committee cache fails, log instead of return err. It doesn't have to be / should not be consensus breaking. (**non-cosmetic**, please review)
 - Rename `InitializeEpochValidators` to `InitializePrecomputeValidators` (**cosmetic**)
 - Align input argument to `beaconState` instead of `st` or `state` (**cosmetic**)
 - Change `st.NumValidators() > len(inactivityScores)` to `!=` to be more strict (**cosmetic**)
 - Update a few instances where we return nothing if `core.CurrentEpoch(beaconState) == cfg.GenesisEpoch`, also added more test coverage  (**cosmetic**)

### CODE REVIEW

File: beacon-chain/core/altair/epoch_spec.go [hunk 1]
```diff
@@ -39,7 +40,7 @@ func ProcessSyncCommitteeUpdates(ctx context.Context, beaconState state.BeaconSt
 			return nil, err
 		}
 		if err := helpers.UpdateSyncCommitteeCache(beaconState); err != nil {
-			return nil, err
+			log.Errorf("Could not update sync committee cache: %v", err)
```
Comment (terencechain) [inline beacon-chain/core/altair/epoch_spec.go:43]: @nisdas lmk if you agree

File: beacon-chain/core/altair/epoch_precompute.go [hunk 2]
```diff
@@ -177,27 +173,28 @@ func ProcessEpochParticipation(
...
+		return beaconState, nil
 	}
 
-	numOfVals := state.NumValidators()
+	numOfVals := beaconState.NumValidators()
 	// Guard against an out-of-bounds using validator balance precompute.
-	if len(vals) != numOfVals || len(vals) != state.BalancesLength() {
-		return state, errors.New("validator registries not the same length as state's validator registries")
+	if len(vals) != numOfVals || len(vals) != beaconState.BalancesLength() {
+		return beaconState, errors.New("validator registries not the same length as beaconState's validator registries")
```
Comment (rkapka) [inline beacon-chain/core/altair/epoch_precompute.go:189]:
```suggestion
		return beaconState, errors.New("validator registries not the same length as state's validator registries")
```

Unwanted refactoring side-effect.

File: beacon-chain/core/altair/epoch_precompute_test.go [hunk 3]
```diff
@@ -325,20 +345,20 @@ func TestProcessRewardsAndPenaltiesPrecompute_GenesisEpoch(t *testing.T) {
...
 	s, err := testState()
 	require.NoError(t, err)
-	validators, balance, err := InitializeEpochValidators(context.Background(), s)
+	validators, balance, err := InitializePrecomputeValidators(context.Background(), s)
 	require.NoError(t, err)
 	_, balance, err = ProcessEpochParticipation(context.Background(), s, balance, validators)
 	require.NoError(t, err)
 	_, err = ProcessRewardsAndPenaltiesPrecompute(s, balance, []*precompute.Validator{})
-	require.ErrorContains(t, "validator registries not the same length as state's validator registries", err)
+	require.ErrorContains(t, "validator registries not the same length as beaconState's validator registries", err)
```
Comment (rkapka) [inline beacon-chain/core/altair/epoch_precompute_test.go:353]:
```suggestion
	require.ErrorContains(t, "validator registries not the same length as state's validator registries", err)
```

Refactoring side-effect.

**Comment by rauljordan** [comment]
@nisdas PTAL

File: beacon-chain/core/altair/epoch_spec.go [hunk 4]
```diff
@@ -35,7 +36,7 @@ func ProcessSyncCommitteeUpdates(ctx context.Context, beaconState state.BeaconSt
...
```
Comment (prestonvanloon) [inline beacon-chain/core/altair/epoch_spec.go:39]: Please use `WithError`

```suggestion
			log.WithError(err).Error("Could not update sync committee cache")
```

**Review by nisdas — DISMISSED** [review]

**Review by rauljordan — APPROVED** [review]

**Pull request merged**